lib/rex/proto/http/client.rb

```diff
@@ -234,7 +234,7 @@ def _send_recv(req, t = -1, persist = false)
 
     send_request(req, t)
 
-    res = read_response(t)
+    res = read_response(req, t)
```

Comment:
I think if we put the req into `opts` this would be a backwards compatible/less risky change 👀 

```suggestion
    res = read_response(t, { original_request: req })
```

Reply:
Good call - I've made this change.